src/NServiceBus.Core/Audit/AuditProcessingStatisticsBehavior.cs

```diff
@@ -6,12 +6,13 @@
 
     class AuditProcessingStatisticsBehavior : IBehavior<IAuditContext, IAuditContext>
     {
+        internal const string ProcessingStartedKey = "NServiceBus.Auditing.ProcessingStarted";
+
         public Task Invoke(IAuditContext context, Func<IAuditContext, Task> next)
         {
-            if (context.Extensions.TryGet(out ProcessingStatisticsBehavior.State state))
+            if (context.Extensions.TryGet(ProcessingStartedKey, out DateTimeOffset processingStarted))
             {
-                context.AuditMetadata[Headers.ProcessingStarted] = DateTimeOffsetHelper.ToWireFormattedString(state.ProcessingStarted);
-                // We can't take the processing time from the state since we don't know it yet.
+                context.AuditMetadata[Headers.ProcessingStarted] = DateTimeOffsetHelper.ToWireFormattedString(processingStarted);
                 context.AuditMetadata[Headers.ProcessingEnded] = DateTimeOffsetHelper.ToWireFormattedString(DateTimeOffset.UtcNow);
             }
```

Comment:
Given that we embarked on a path to no longer make these statistics optional I wonder if calculating it would be better placed into the AuditToRoutingConnector right before we prevent additional modifications to the context?

Reply:
Are you referring to the `ProcessingEnded` value? I don't mind too much about how exactly we determine these values, I think that's not the crucial point of this PR and could be changed via subsequent PRs too if we feel there are more appropriate ways to handle this.